RecoLocalCalo/HcalRecProducers/src/FacileHcalReconstructor.cc

```diff
@@ -0,0 +1,101 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonClient.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicEDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
+#include "Geometry/Records/interface/HcalRecNumberingRecord.h"
+
+class FacileHcalReconstructor : public SonicEDProducer<TritonClient> {
+public:
+  explicit FacileHcalReconstructor(edm::ParameterSet const& cfg)
+      : SonicEDProducer<TritonClient>(cfg),
+        fChannelInfoName_(cfg.getParameter<edm::InputTag>("ChannelInfoName")),
+        fTokChannelInfo_(consumes<HBHEChannelInfoCollection>(fChannelInfoName_)),
+        htopoToken_(esConsumes<HcalTopology, HcalRecNumberingRecord>()) {
+    produces<HBHERecHitCollection>();
+    setDebugName("FacileHcalReconstructor");
+  }
+
+  void acquire(edm::Event const& iEvent, edm::EventSetup const& iSetup, Input& iInput) override {
+    edm::Handle<HBHEChannelInfoCollection> hChannelInfo;
+    iEvent.getByToken(fTokChannelInfo_, hChannelInfo);
+
+    const HcalTopology* htopo = &iSetup.getData(htopoToken_);
+
+    auto& input1 = iInput.begin()->second;
+    auto data1 = std::make_shared<TritonInput<float>>();
+    data1->reserve(hChannelInfo->size());
+    client_.setBatchSize(hChannelInfo->size());
+
+    hcalIds_.clear();
+
+    for (const auto& pChannel : *hChannelInfo) {
+      std::vector<float> input;
+      const HcalDetId pDetId = pChannel.id();
+      hcalIds_.push_back(pDetId);
+
+      //FACILE uses iphi as a continuous variable
+      input.push_back((float)pDetId.iphi());
+      input.push_back((float)pChannel.tsGain(0.));
+      for (unsigned int itTS = 0; itTS < pChannel.nSamples(); ++itTS) {
+        input.push_back((float)pChannel.tsRawCharge(itTS));
+      }
+
+      //FACILE considers 7 Hcal depths as binary variables
+      for (int itDepth = 1; itDepth <= htopo->maxDepth(); itDepth++) {
+        input.push_back(pDetId.depth() == itDepth);
+      }
+
+      //ieta is also encoded as a binary variable
+      for (int itIeta = 0; itIeta <= htopo->lastHERing(); itIeta++) {
+        input.push_back(pDetId.ieta() == itIeta);
```

Comment:
possible bug: before, this was `abs(ieta)`
actually, the `ietaAbs()` function of `HcalDetId` can be used